moler/cmd/unix/tail_latest_file.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+"""
+Tail latest file from the  directory.
+"""
+
+import os
+import time
+from moler.cmd.unix.genericunix import GenericUnixCommand
+
+__author__ = 'Tomasz Krol'
+__copyright__ = 'Copyright (C) 2020, Nokia'
+__email__ = 'tomasz.krol@nokia.com'
+
+
+class TailLatestFile(GenericUnixCommand):
+    def __init__(self, connection, directory, file_pattern="*", prompt=None, newline_chars=None, runner=None,
+                 time_for_failure=0.1):
+        """
+        Command for tail latest file from the directory.
+        :param connection: Moler connection to device, terminal when command is executed.
+        :param directory: path to directory to tail.
+        :param file_pattern: pattern for files from directory.
+        :param prompt: prompt (on system where command runs).
+        :param newline_chars: Characters to split lines - list.
+        :param runner: Runner to run command.
+        :param time_for_failure: time (in seconds) for failure indication from first line of output. Set to 0 if skip
+        all failure indications.
+        """
+        super(TailLatestFile, self).__init__(connection=connection, prompt=prompt, newline_chars=newline_chars,
+                                             runner=runner)
+
+        self.directory = directory
+        self.file_pattern = file_pattern
+        self.ret_required = False
+        self.time_for_failure = time_for_failure
+        self._first_line_time = None
+        self._check_failure_indication = True
+
+        self._max_index_from_beginning = 0  # Right (from 0 to this) index of substring of command_string passed
+        self._max_index_from_end = 0  # Left (from this to the end) index of substring of command_string passed
+        self._multiline_cmd = True
+
+    def build_command_string(self):
+        """
+        Build command string from parameters passed to object.
+
+        :return: String representation of command to send over connection to device.
+        """
+
+        file_path = os.path.join(self.directory, self.file_pattern)
+        bash_script = "bash -c '\n" \
+                      'echo "Press [CTRL+C] to stop.."\n' \
+                      r'trap "kill \$tail_pid; exit" INT' \
+                      '\n' \
+                      'last_file=""\n' \
+                      'tail_pid=""\n' \
+                      'while :\n' \
+                      'do\n' \
+                      'current_last_file=`ls -t {} | head -1`\n' \
+                      'if [ "$last_file" != "$current_last_file" ]\n' \
+                      'then\n' \
+                      '[ -n "$tail_pid" ] && kill $tail_pid\n' \
+                      'last_file=$current_last_file\n' \
+                      'tail -f $last_file &\n' \
```

Comment:
`tail -f $current_last_file` takes only 10 last lines (if file has more lines then they will not be shown) `tail -f -n +1 $current_last_file` should show all lines

Reply:
Done